codegen/lib/templates/kotlin/android_class.erb

```diff
@@ -60,4 +69,44 @@ actual class <%= entity.name %> private constructor(
 <%- end -%>
     }
 <%  end -%>
+
+<%  unless entity.methods.select{ |x| x.name == "Delete" }.empty? -%>
+    internal class <%= entity.name %>PhantomReference private constructor(
+        referent: <%= entity.name %>,
+        private val nativeHandle: Long,
+    ) :
+        java.lang.ref.PhantomReference<<%= entity.name %>>(referent, queue) {
+        companion object {
+            private val references: MutableSet<<%= entity.name %>PhantomReference> = HashSet()
+            private val queue: java.lang.ref.ReferenceQueue<<%= entity.name %>> = java.lang.ref.ReferenceQueue<<%= entity.name %>>()
+
+            init {
+                Thread {
+                    try {
+                        doDeletes()
+                    } catch (e: InterruptedException) {
```

Comment:
Do we need to catch other exceptions? If something goes wrong and another exception will be thrown we would still get a leak as reference might be still there

Reply:
don't expect any exception to be thrown there, what do you think can be?

Reply:
don't have anything in particular in mind, was thinking theoretically about native code exceptions